src/services/reminder-service.ts

```diff
@@ -0,0 +1,22 @@
+import { isUserTarget, Reminder } from "../models/reminder";
+import { Collections, db } from "./database"
+import { WebSocket } from "../app";
+
+export class ReminderService {
+  static async initSendReminder(interval: number): Promise<NodeJS.Timeout> {
+    return setInterval(async () =>{
+   
+    const reminders: Reminder[] = await db.collection(Collections.reminders).get().then((snapshot) => (snapshot.docs.map((d) => { const data = d.data(); data.id = d.id; return data as Reminder })));
+    for (const reminder of reminders) {
+      const time = new Date(reminder.date);
+      if (time.getTime() < Date.now()){
+        if(isUserTarget(reminder.target)){
+          //TODO send to random bot to send dm
```

Comment:
not really random, send it to the correct shard but the socket will do that if you provide it the guild id

Reply:
correction, user dm's should always be on shard 0